### PR TITLE
add elpd dashboard

### PR DESCRIPTION
**Pull request by yilinxia**
add elpd dashboard (part of GSoC 2022) and modify the readme file to provide more guidlines for future development

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,37 @@
-# arviz_dashboard
-Bayesian Dashboards built on top of Panel
```
Comment (aloctavodia) [inline README.md:2]: Exploratory analysis of Bayesian models with dashboards

File: README.md [hunk 2]
```diff
@@ -1,4 +1,37 @@
...
+# Declarative Dashboard for Arviz
 
-See [wiki](https://github.com/arviz-devs/arviz_dashboard/wiki)
+## Project Overview
+This project intends to incorporate all sorts of ArviZ graphs in the dashboard as a separate Python library or ArviZ module which enable users compare different visualizations in the same view and interact with different visualizations at the same time.
```
Comment (aloctavodia) [inline README.md:4]:
```suggestion
This project brings dashboards to ArviZ enabling users to compare different visualizations in the same view and interact with them. 
```

File: README.md [hunk 3]
```diff
@@ -1,4 +1,37 @@
-# arviz_dashboard
-Bayesian Dashboards built on top of Panel
+# Declarative Dashboard for Arviz
 
-See [wiki](https://github.com/arviz-devs/arviz_dashboard/wiki)
+## Project Overview
+This project intends to incorporate all sorts of ArviZ graphs in the dashboard as a separate Python library or ArviZ module which enable users compare different visualizations in the same view and interact with different visualizations at the same time. 
+
+We have successfully implemented the first dashboard on ELPD and model comparison,which you can find in the `examples/elpd_dashboard.ipynb`. In the ear future, more dashboards will be added to this repo.
```
Comment (aloctavodia) [inline README.md:6]:
```suggestion
We currently provide one dashboard for model comparison, in terms of the ELPD_loo values, which you can find in the `examples/elpd_dashboard.ipynb`. In the ear future, more dashboards will be added to this repo.
```

File: environment.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+name: arviz_dashboard
+channels:
+  - conda-forge
+  - defaults
+dependencies:
+  - python=3.8
+  - arviz
+  - jupyterlab
+  - panel
+  - pymc3
```
Comment (aloctavodia) [inline environment.yml:10]: pymc should not be a dependency. The same applies to other ppls

**Review by aloctavodia — COMMENTED** [review]
We should have a requirement file and a setup instead of an environment file

File: README.md [hunk 5]
```diff
@@ -1,4 +1,37 @@
...
+- Dashboard Components [Panel Widgets](https://panel.holoviz.org/user_guide/Widgets.html)
+- Dashboard Interaction [Panel Depends](https://panel.holoviz.org/user_guide/APIs.html) , [Panel Interact](https://panel.holoviz.org/user_guide/Interact.html), [Panel Link](https://panel.holoviz.org/user_guide/Links.html)
+- Jupyter Widget [ipywidget](https://ipywidgets.readthedocs.io/en/stable/) 
+
+We should think about transforming the entire dashboard into a template to process various inputs for future development. However, we haven't figured out how to achieve that and it would be advisable to attempt the following options.
+
+- Containerize the whole dashboard as a class and different models with inference data as the input
+- Derive ideas from [Panel Templates](https://panel.holoviz.org/user_guide/Templates.html)
+
+Additionally, we need to consider how the dashboard will be shown. We currently have a dashboard embedded within a Jupyter lab (notebook). However, it is important to consider if the dashboard should be used as an extension to Jupyter or as a standalone website.
```
Comment (aloctavodia) [inline README.md:33]: I think having dashboard should be functions like other arviz functions and they should be used together as part on an integrated analysis, so I don't see the appealing of a standalone website. BUt again we can keep discussing this on its own issue (or on slack)

File: elpd_dashboard.ipynb [hunk 6]
```diff
@@ -0,0 +1,811 @@
+{
```
Comment (aloctavodia) [inline elpd_dashboard.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #10.    y_max_fi = pn.widgets.FloatInput(name='y_max', value=0, step=0.1, start=-100, end=100, width=200)
</pre><p>Maybe we can just remove this controls and let matplotlib to automatically adjust the y.axis?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/arviz-devs/arviz_dashboard/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (aloctavodia) [inline elpd_dashboard.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #15.    global idatas_cmp
</pre><p>There is a way to remove these globals?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/arviz-devs/arviz_dashboard/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (aloctavodia) [inline elpd_dashboard.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #146.    pn.Column(pn.Row(model_selection1,model_selection2),
</pre><p>Enclose this into <code>display()</code>and all the content of this notebook into a function. Then calling the function will generate the dashboard.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/arviz-devs/arviz_dashboard/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

**Comment by aloctavodia** [comment]
Can you move most of the content of the notebook to a .py file and in the notebook just import and call the function?

**Comment by yilinxia** [comment]
> 

have changed the file structure accordingly

**Review by aloctavodia — APPROVED** [review]
LGTM!

**Pull request merged**